test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/ProjectSystems/LegacyPackageReferenceRestoreUtilityTests.cs

```diff
@@ -893,6 +893,143 @@ public async Task DependencyGraphRestoreUtility_LegacyPackageRef_Restore_Package
+                        testSolutionManager,
+                        dgSpec,
+                        restoreContext,
+                        new RestoreCommandProvidersCache(),
+                        (c) => { },
+                        sourceRepositoryProvider.GetRepositories(),
+                        Guid.Empty,
+                        false,
+                        true,
+                        testLogger,
+                        CancellationToken.None);
+
+                    foreach (var restoreSummary in restoreSummaries)
+                    {
+                        Assert.True(restoreSummary.Success);
+                        Assert.False(restoreSummary.NoOpRestore);
+                    }
+
+                    Assert.True(File.Exists(projectLockFilePath));
+
+                    // delete existing restore output files
+                    File.Delete(Path.Combine(vsProjectAdapter.MSBuildProjectExtensionsPath, "project.assets.json"));
+
+                    // clean packages folder
+                    Directory.Delete(testSolutionManager.GlobalPackagesFolder, true);
+                    Directory.CreateDirectory(testSolutionManager.GlobalPackagesFolder);
+
+                    // overwrite the package
+                    FileUtility.Delete(packageSource);
+                    var newPackageA = new SimpleTestPackageContext("packageA", "1.0.0");
+                    newPackageA.AddFile("lib/net45/a1.dll");
+                    var newPackageB = new SimpleTestPackageContext("packageB", "1.0.0");
+                    newPackageB.AddFile("lib/net45/b1.dll");
+                    SimpleTestPackageUtility.CreateOPCPackage(newPackageA, packageSource);
+                    SimpleTestPackageUtility.CreateOPCPackage(newPackageB, packageSource);
+
+                    // Act
+                    restoreSummaries = await DependencyGraphRestoreUtility.RestoreAsync(
+                        testSolutionManager,
+                        dgSpec,
+                        restoreContext,
+                        new RestoreCommandProvidersCache(),
+                        (c) => { },
+                        sourceRepositoryProvider.GetRepositories(),
+                        Guid.Empty,
+                        false,
+                        true,
+                        testLogger,
+                        CancellationToken.None);
+
+                    // Assert
+                    foreach (var restoreSummary in restoreSummaries)
+                    {
+                        Assert.False(restoreSummary.Success);
+                        Assert.True(restoreSummary.Errors.Count > 0);
+                        var nu1403Message = restoreSummary.Errors.FirstOrDefault(message => (message as RestoreLogMessage).Code == NuGetLogCode.NU1403);
+                        Assert.NotNull(nu1403Message);
+                        Assert.Contains("packageA.1.0.0", nu1403Message.Message);
```

Comment:
Check for hashes too?

Reply:
Fixed.

Reply:
Is this resolved?  I don't see the test checking for expected and actual hashes.

Reply:
I didn't add the actual hashes in message verification, just that the build of the message is there. 
I can add the hashes too. Originally I opted against it, as the APIs for that are not well exposed, but it shouldn't be too bad, given the packages used are not signed.